Optimize & Consolidate Bundle Sidebar & Bundle View

leilenah opened the pull request:
### Reasons for making this change

This change optimizes the UI for viewing bundle data in the worksheet view as well as the standalone bundle view. Now both views leverage the same `BundleDetail` component, which has been improved from a UI perspective. 

### Related issues

https://github.com/codalab/codalab-worksheets/issues/4131

### Bundle Detail Demo

https://user-images.githubusercontent.com/25855750/180516127-4667175e-a5eb-4570-970f-1be428750f4e.mov


### Bundle View Demo

https://user-images.githubusercontent.com/25855750/180516888-a2b52eec-d37b-4e78-aeb7-17d3eeb2c722.mov


### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [ ] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [ ] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

leilenah committed:
Bundle View prototype v1.

leilenah committed:
Merge branch 'master' into feature/4131-bundle-view-ui

leilenah committed:
Optimize bundle detail sidebar and bundle view

leilenah committed:
Fix getTimePassed in state table.

leilenah committed:
Update Worksheet snapshot.

leilenah committed:
Merge branch 'master' into feature/4131-bundle-view-ui

leilenah committed:
Update failing snapshots.

leilenah committed:
Remove commented code.

percyliang reviewed (commented):
This is great - much cleaner and more uniform and helpful!

Just a few minor comments:
1. The help icon next to each field is very faint.
2. UUID maybe should go at the top before name.
3. Bundle page could appear in the upper right corner like a NE arrow that looks like it'll expand things.
4. Store should be after Size, logically
5. Allowed Failed, Exclude Patterns can go after all the Request fields.
6. The fields could better match the rendering when you create a new run (especially the grouping of the resources) - could we share a component?
7. In the help text for State, we should write the state again (e.g., "staged: ...") so it's clear we're talking about the highlighted state.  Maybe when you mouse over the states, they should show a description too.

epicfaace commented:
My feedback: https://github.com/codalab/codalab-worksheets/issues/4153

leilenah committed:
Bundle view refactors.

leilenah committed:
Add kill confirmation for bundle rows.

leilenah committed:
Conditional render bundle deps table.

leilenah committed:
Refine MoreDetail rendering based on edit permissions

leilenah committed:
Merge branch 'master' into feature/4131-bundle-view-ui

leilenah commented:
@percyliang 

> The fields could better match the rendering when you create a new run (especially the grouping of the resources) - could we share a component?

I have modified the labels to align with the labels in the new run form. I have also added field groupings. 

> In the help text for State, we should write the state again (e.g., "staged: ...") so it's clear we're talking about the highlighted state. Maybe when you mouse over the states, they should show a description too.

Note that since `state_details` are provided by the rest API for a single bundle, that bundle won't have the details for every state in the bundle lifecycle. It only has the details for the current state. I have added a tooltip for the current state in the lifecycle diagram, but the inactive states don't have tooltips. I could hardcode the definitions to achieve this, but I wanted to avoid that.

percyliang reviewed (approved):
Great work!

leilenah committed:
Move bundle state times to right of bundle box

leilenah committed:
Update code docs

leilenah committed:
Fix formatting

leilenah committed:
Lighten state time color.
